src/arewefluentyet/milestone.py

```diff
@@ -0,0 +1,74 @@
+import os
+import json
+from datetime import date
+
+
+def parse_date(input):
+    return date(*(int(s) for s in input.split("-")))
+
+
+class Milestone:
+    name = None
+    start_date = None
+    data_path = None
+    progress_data = None
+
+    def __init__(self, data_path):
+        self.data_path = os.path.join(data_path, self.name)
+
+    def append_progress_entry(self, progress_entry):
+        self.get_progress_data().append(progress_entry)
+
+    def get_progress_data(self):
+        if self.progress_data is None:
+            path = os.path.join(self.data_path, "progress.json")
+
+            if os.path.exists(path):
+                self.progress_data = json.load(open(path))
+            else:
+                self.progress_data = []
+
+        return self.progress_data
+
+    def has_log_for_date(self, source, date):
```

Comment:
Can you add a comment on what this does, and why `return False` is a good default behavior?